src/melee.cpp

```diff
@@ -1940,3 +1945,129 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
+
+    const char *target_name = target.name.c_str();
+    const char *item_name = it.tname().c_str();
```

Comment:
UB. `tname()` returns `std::string` which gets destroyed right after the assignment. The pointer returned by `c_str()` is invalid after that. The line above is fine (`target` and its `name` live on), but I'd modify it too, just in case.